### PR TITLE
Fix11

### PR DESCRIPTION
**Pull request by HansOlsson**
Since we repeatedly find cases where errors that have remained from version 1.1 of the specification I'm going through that version and looking for problems.

I also found that:
https://github.com/modelica/ModelicaSpecification/blob/07ccecd0719e130ad7edfb578f94e72ba5872dbe/chapters/equations.tex#L106
is problematic ("once"? what if the for-equation is nested?), but I couldn't find any good alternative.

### CODE REVIEW

File: chapters/connectors.tex [hunk 1]
```diff
@@ -384,7 +384,7 @@ \section{Generation of Connection Equations}\label{generation-of-connection-equa
 \begin{lstlisting}[language=modelica]
 connect(a, b);
 \end{lstlisting}
-the primitive components of \lstinline!a! and \lstinline!b! form a connection set, together with an indication of whether they are from an inside or an outside connector.
+each of their primitive components will generate one connection set comprised of the corresponding primitive components of \lstinline!a! and \lstinline!b! together with an indication of whether they are from an inside or an outside connector.
```
Comment (eshmoylova) [inline chapters/connectors.tex:387]: This is repetitive: primitive components will generate one ... set ...comprised of ... primitive components. Perhaps:
```suggestion
one connection set is generated for every pair of corresponding primitive components of \lstinline!a! and \lstinline!b! together with an indication of whether they are from an inside or an outside connector.
```

**Review by eshmoylova — CHANGES_REQUESTED** [review]

File: chapters/connectors.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/connectors.tex:387]: Ah, yes, that's better.

**Review by eshmoylova — APPROVED** [review]

File: chapters/arrays.tex [hunk 2]
```diff
@@ -1447,14 +1447,16 @@ \section{Empty Arrays}\label{empty-arrays}
 \begin{lstlisting}[language=modelica]
 Real[0, m] * Real[m, n] = Real[0, n] // empty matrix
 Real[m, n] * Real[n, 0] = Real[m, 0] // empty matrix
-Real[m, 0] * Real[0, n] = fill(0.0, m, n) // non-empty matrix of zeros
+Real[m, 0] * Real[0, n] = fill(0.0, m, n) // matrix of zeros
+// Note fill(0.0, m, n) will be an empty matrix if m or n is 0
 \end{lstlisting}
```
Comment (henrikt-ma) [inline chapters/arrays.tex:1452]: Nicer with the explanation as normal text:
```suggestion
\end{lstlisting}
Note that \lstinline!fill(0.0, m, n)! will be an empty matrix if \lstinline!m! or \lstinline!n! is zero.
```

File: chapters/connectors.tex [hunk 3]
```diff
@@ -384,7 +384,7 @@ \section{Generation of Connection Equations}\label{generation-of-connection-equa
 \begin{lstlisting}[language=modelica]
 connect(a, b);
 \end{lstlisting}
-the primitive components of \lstinline!a! and \lstinline!b! form a connection set, together with an indication of whether they are from an inside or an outside connector.
+one connection set is generated for every pair of corresponding primitive components of \lstinline!a! and \lstinline!b! together with an indication of whether they are from an inside or an outside connector.
```
Comment (henrikt-ma) [inline chapters/connectors.tex:387]: Tone down emphasis on the count:
```suggestion
a connection set is generated for every pair of corresponding primitive components of \lstinline!a! and \lstinline!b! together with an indication of whether they are from an inside or an outside connector.
```

File: chapters/functions.tex [hunk 4]
```diff
@@ -882,10 +882,8 @@ \subsection{Automatic Vectorization}\label{scalar-functions-applied-to-array-arg
   sum1 := e1 + e2;
 end Add;
 \end{lstlisting}
-\lstinline!Add(1, [1,2,3])! adds one to each of the elements of the second
-argument giving the result \lstinline![2,3,4]!. However, it is illegal to
-write \lstinline!1 + [1,2,3]!, because the rules for the built-in
-operators are more restrictive.
+\lstinline!Add(1, [1,2,3])! adds one to each of the elements of the second argument giving the result \lstinline![2,3,4]!.
```
Comment (henrikt-ma) [inline chapters/functions.tex:885]: Should we take the opportunity to also change to the common convention of lower camel case for function classes, `Add` → `add`?

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
I like the initiative to go back and find old mistakes.  Just some minor suggestions.

File: chapters/connectors.tex (hunk 3 above)
Comment (HansOlsson) [reply chapters/connectors.tex:387]: I'm not sure which variant is best; I can understand that being clear that it is one for each pair might avoid ambiguities.

Comment (henrikt-ma) [reply chapters/connectors.tex:387]: Maybe _a… each_ is better?
```suggestion
a connection set is generated for each pair of corresponding primitive components of \lstinline!a! and \lstinline!b! together with an indication of whether they are from an inside or an outside connector.
```

File: chapters/functions.tex (hunk 4 above)
Comment (HansOlsson) [reply chapters/functions.tex:885]: Yes. Changing.

Comment (henrikt-ma) [reply chapters/functions.tex]: OK. Resolving.

**Comment by HansOlsson** [comment]
@henrikt-ma ok now?

**Pull request merged**